framework/core/js/src/admin/components/PermissionDropdown.tsx

```diff
@@ -89,40 +94,39 @@ export default class PermissionDropdown extends Dropdown {
           {
             icon: !everyone && !members ? 'fas fa-check' : true,
             disabled: !everyone && !members,
-            onclick: (e) => {
+            onclick: (e: MouseEvent) => {
               if (e.shiftKey) e.stopPropagation();
               this.save([]);
             },
           },
-          [badgeForId(adminGroup.id()), ' ', adminGroup.namePlural()]
+          [badgeForId(adminGroup.id()!), ' ', adminGroup.namePlural()]
         )
       );
 
-      [].push.apply(
-        children,
-        app.store
-          .all('groups')
-          .filter((group) => [Group.ADMINISTRATOR_ID, Group.GUEST_ID, Group.MEMBER_ID].indexOf(group.id()) === -1)
-          .map((group) =>
-            Button.component(
-              {
-                icon: groupIds.indexOf(group.id()) !== -1 ? 'fas fa-check' : true,
-                onclick: (e) => {
-                  if (e.shiftKey) e.stopPropagation();
-                  this.toggle(group.id());
-                },
-                disabled: this.isGroupDisabled(group.id()) && this.isGroupDisabled(Group.MEMBER_ID) && this.isGroupDisabled(Group.GUEST_ID),
+      const groupButtons = app.store
+        .all<Group>('groups')
+        .filter((group) => ![Group.ADMINISTRATOR_ID, Group.GUEST_ID, Group.MEMBER_ID].includes(group.id()!))
+        .map((group) =>
+          Button.component(
```

Comment:
JSX would be more consistent with our code style, albeit more verbose

Reply:
that's taken care of in #3609